In [42]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline
from sklearn.preprocessing import StandardScaler

df = pd.read_csv('cleanedData.csv')
df.head(5)

City    State         Zip  Aspergillus flavus/oryzae  \
0  Mountain Brook  Alabama       35223                        0.0   
1         Wasilla   Alaska       99623                        0.0   
2        Soldotna   Alaska  99669-8654                        0.0   
3        Soldotna   Alaska       99669                        0.0   
4       Anchorage   Alaska  99508-4014                       64.0   

   Aspergillus fumigatus  Aspergillus niger  Aspergillus ochraceus  \
0                    0.0               14.0                    0.0   
1                    0.0               12.0                    0.0   
2                    3.0                3.0                  234.0   
3                    0.0                0.0                    0.0   
4                    5.0               32.0                    0.0   

   Aspergillus penicillioides  Aspergillus restrictus*  \
0                         0.0                     11.0   
1                         5.0                      0.0   
2                         0.0                      6.0   
3                         0.0                      0.0   
4                         0.0                      6.0   

   Aspergillus sclerotiorum  ...  Immune_chronicmonoorepstein-barrvirus  \
0                       0.0  ...                                    0.0   
1                       0.0  ...                                    0.0   
2                       0.0  ...                                    1.0   
3                       0.0  ...                                    0.0   
4                       0.0  ...                                    0.0   

   Immune_frequentherpesoutbreaks  Immune_viralinfectionsbecomebacterial  \
0                             0.0                                    0.0   
1                             0.0                                    0.0   
2                             0.0                                    0.0   
3                             0.0                                    0.0   
4                             0.0                                    0.0   

   Immune_increasedsusceptibilitytocancer  \
0                                     0.0   
1                                     0.0   
2                                     0.0   
3                                     0.0   
4                                     0.0   

   Reproductive_changesinmenstrualcycle  \
0                                   1.0   
1                                   0.0   
2                                   1.0   
3                                   0.0   
4                                   0.0   

   Reproductive_inconsistentmenstrualcycle  \
0                                      1.0   
1                                      0.0   
2                                      1.0   
3                                      0.0   
4                                      0.0   

   Reproductive_vaginalyeastorbacterialinfections  Reproductive_jockitch  \
0                                             1.0                    1.0   
1                                             1.0                    0.0   
2                                             0.0                    0.0   
3                                             0.0                    0.0   
4                                             0.0                    0.0   

   Reproductive_infertilityinbothgenders  Reproductive_hormoneimbalances  
0                                    1.0                             1.0  
1                                    0.0                             1.0  
2                                    0.0                             1.0  
3                                    0.0                             0.0  
4                                    0.0                             0.0  

[5 rows x 145 columns]

Principal Component Analysis(PCA) to reduce the 145 columns

In [43]:
#Inital Shape
df.shape

(581, 145)

In [44]:
# Extracting mold columns for PCA analysis
moldDF = df.loc[:,'Aspergillus flavus/oryzae':'Rhizopus stolonifer']
moldDF.head()

Aspergillus flavus/oryzae  Aspergillus fumigatus  Aspergillus niger  \
0                        0.0                    0.0               14.0   
1                        0.0                    0.0               12.0   
2                        0.0                    3.0                3.0   
3                        0.0                    0.0                0.0   
4                       64.0                    5.0               32.0   

   Aspergillus ochraceus  Aspergillus penicillioides  Aspergillus restrictus*  \
0                    0.0                         0.0                     11.0   
1                    0.0                         5.0                      0.0   
2                  234.0                         0.0                      6.0   
3                    0.0                         0.0                      0.0   
4                    0.0                         0.0                      6.0   

   Aspergillus sclerotiorum  Aspergillus sydowii  Aspergillus unguis  \
0                       0.0                  0.0                 0.0   
1                       0.0                  0.0                 0.0   
2                       0.0                  5.0                 0.0   
3                       0.0                  0.0                 0.0   
4                       0.0                  0.0                 0.0   

   Aspergillus versicolor  ...  Acremonium strictum  Alternaria alternata  \
0                     0.0  ...                  9.0                  19.0   
1                     0.0  ...                  0.0                   0.0   
2                   279.0  ...                  0.0                   0.0   
3                     6.0  ...                  0.0                   0.0   
4                     6.0  ...                  2.0                   3.0   

   Aspergillus ustus  Cladosporium cladosporioides 1  \
0               15.0                             0.0   
1                0.0                           470.0   
2              119.0                             0.0   
3                0.0                             0.0   
4                0.0                             0.0   

   Cladosporium cladosporioides 2  Cladosporium herbarum  Epicoccum nigrum  \
0                             0.0                    0.0             372.0   
1                            12.0                  114.0              67.0   
2                             0.0                    0.0              60.0   
3                             0.0                    0.0             435.0   
4                             0.0                    0.0               2.0   

   Mucor amphibiorum*  Penicillium chrysogenum  Rhizopus stolonifer  
0                10.0                     29.0                  6.0  
1                24.0                    111.0                  0.0  
2               173.0                     70.0                  0.0  
3                 4.0                     49.0                  0.0  
4                 7.0                     83.0                 12.0  

[5 rows x 36 columns]

In [ ]:
#Scaling data, normalizing it to a common scale using standard 

scaler = StandardScaler()
scaler.fit(moldDF)
scaled_data = scaler.transform(moldDF)